graylog2-server/src/test/java/org/graylog2/inputs/persistence/MongoInputStatusServiceTest.java

```diff
@@ -182,24 +188,25 @@ public void delete_ReturnsZero_WhenDeletingNonExistantRecord() {
 
     @Test
     @MongoDBFixtures("input-status.json")
-    public void handleDeleteEvent_DoesNothing() {
-        /*
-        Currently, the InputDeleted event is propagated both when an input is stopped and when an input is deleted. We
-        would like to clean up the DB when an input is deleted, but not when it is stopped.  This method is in place to
-        be used once there are separate events for input deleted and input stopped.  For now, it should do nothing.
-         */
-
-        cut.handleInputDeleted(new InputDeleted(){
+    public void handleDeleteEvent_DoesNothing() throws Exception {
+        final String deletedInput = "54e3deadbeefdeadbeef0001";
+        final InputDeleted inputDeletedEvent = new InputDeleted() {
             @Override
             public String id() {
-                return "54e3deadbeefdeadbeef0001";
+                return deletedInput;
             }
-        });
+        };
 
+        cut.handleInputDeleted(inputDeletedEvent);
         // The record should not be removed from the DB
-        Optional<InputStatusRecord> optDbRecord = cut.get("54e3deadbeefdeadbeef0001");
+        assertThat(cut.get(deletedInput).isPresent(), is(true));
 
-        assertThat(optDbRecord, notNullValue());
-        assertThat(optDbRecord.isPresent(), is(true));
+        // Simulate that the input has actually been deleted
+        // TODO: This will change once we fix https://github.com/Graylog2/graylog2-server/issues/7812
+        when(inputService.find(deletedInput)).thenThrow(new NotFoundException());
+
+        cut.handleInputDeleted(inputDeletedEvent);
+        // The record should be removed from the DB
+        assertThat(cut.get(deletedInput).isPresent(), is(false));
     }
```

Comment:
This should be split into two separate tests to make it easier to see if a change breaks one but not the other.

Reply:
Good point! :+1: